examples/flame_simulation.py

```diff
@@ -0,0 +1,113 @@
+### Flame simulation on the Microbit.
+### Author: M. Schafer 2016
+# This program has been placed into the public domain.
+
+import microbit, random
+
+# User adjustable values for range of brightness in flames.
+min_brightness = 2
+max_brightness = 8
+
+
+#  fixed for the Microbit
+display_width  = 5
+display_height = 5
+
+invert_display = True # flame can be oriented in either direction
+
+# mask to create fire shape. multiplies values %
+Mask = [[ 88, 100, 100, 100, 88 ],
+        [ 60,  95, 100,  95, 60 ], 
+        [ 50,  88,  90,  88, 50 ], 
+        [ 33,  75,  88,  75, 33 ],
+        [ 10,  33,  66,  33, 10 ]  ]
+
+# Generate a new bottom row of random values for the flames
+def generate_line(line, start=min_brightness, end=max_brightness):
+    "start and end define range of dimmest to brightest 'flames'"
+    for i in range(display_width):
```

Comment:
This would be nicer if you returned a new list: `return [start + random.randrange(end-start) for i in range(display_width)]`